watertap/costing/costing_base.py

```diff
@@ -35,6 +35,188 @@ class WaterTAPCostingBlockData(FlowsheetCostingBlockData):
+            )
+            / (
+                pyo.units.convert(
+                    flow_rate, to_units=pyo.units.m**3 / self.base_period
+                )
+                * self.utilization_factor
+            ),
+            doc=f"Constraint for Levelized Cost of Water based on flow {flow_rate.name}",
+        )
+        self.add_component(name + "_constraint", LCOW_constraint)
+
+        self._registered_LCOWs[name] = (LCOW, LCOW_constraint)
+
+    def add_specific_energy_consumption(
+        self, flow_rate, name="specific_energy_consumption"
+    ):
+        """
+        Add specific energy consumption (kWh/m**3) to costing block.
+        Args:
+            flow_rate - flow rate of water (volumetric) to be used in
+                        calculating specific energy consumption
+            name (optional) - the name of the Expression for the specific
+                              energy consumption (default: specific_energy_consumption)
+        """
+
+        self.add_component(
+            name,
+            pyo.Expression(
+                expr=self.aggregate_flow_electricity
+                / pyo.units.convert(
+                    flow_rate, to_units=pyo.units.m**3 / pyo.units.hr
+                ),
+                doc=f"Specific energy consumption based on flow {flow_rate.name}",
+            ),
+        )
+
+    def add_annual_water_production(self, flow_rate, name="annual_water_production"):
+        """
+        Add annual water production to costing block.
+        Args:
+            flow_rate - flow rate of water (volumetric) to be used in
+                        calculating annual water production
+            name (optional) - name for the annual water productionvariable
+                              Expression (default: annual_water_production)
+        """
+        self.add_component(
+            name,
+            pyo.Expression(
+                expr=(
+                    pyo.units.convert(
+                        flow_rate, to_units=pyo.units.m**3 / self.base_period
+                    )
+                    * self.utilization_factor
+                ),
+                doc=f"Annual water production based on flow {flow_rate.name}",
+            ),
+        )
+
+    def add_electricity_intensity(self, flow_rate, name="electricity_intensity"):
+        """
+        Add calculation of overall electricity intensity to costing block.
+        Args:
+            flow_rate - flow rate of water (volumetric) to be used in
+                        calculating electricity intensity
+            name (optional) - the name of the Expression for the specific
+                              electrical intensity (default: specific_electrical_carbon_intensity)
+        """
+        self.add_specific_energy_consumption(flow_rate, name=name)
+
+    def add_specific_electrical_carbon_intensity(
+        self, flow_rate, name="specific_electrical_carbon_intensity"
+    ):
+        """
```

Comment:
As we consolidate packages, we may want to consider using wordage that aligns with the national decarbonization effort (e.g., CO2_equivalent_footprint or whatever is ubiquitously used nowadays).